backend/requirements.txt

```diff
@@ -4,8 +4,15 @@ langchain-huggingface==0.0.3
 langchain-google-genai==1.0.7
 langchain-google-vertexai==1.0.6
 langgraph==0.1.8
-beautifulsoup4==4.12.3
 fastapi==0.111.0
 faiss-cpu==1.8.0
 markdown==3.0.0
 rank-bm25==0.2.2
+sphinx==7.4.4
+sphinx-autobuild==2024.4.16
+myst-parser==3.0.1
+sphinxcontrib-mermaid==0.9.2
+sphinx-book-theme==1.1.3
+sphinx-copybutton==0.5.2
+sphinx-external-toc==1.0.1
+pypdf==4.2.0
```

Comment:
is Pypdf accurate? How does it compare to this text-based file

https://github.com/The-OpenROAD-Project/OpenSTA/blob/master/doc/OpenSTA.odt